### PR TITLE
Adds optional arguments to gmsh init

### LINKED ISSUE #434: Using pygmsh alters system path on win32
running the following code snippet

```python
import pygmsh

l,w,h = 5, 10, 4

vertices = [
            (0, 0, 0),
            (w, 0, 0),
            (w, h, 0),
            (0, h, 0),
            (0, h, l),
            (w, h, l),
            (w, 0, l),
            (0, 0, l),
            ]

faces = [
            (0, 1, 2, 3),  #face front
            (2, 3, 4, 5),  #face top
            (1, 2, 5, 6),  #face right 
            (0, 7, 4, 3),  #face left
            (5, 4, 7, 6),  #face back
            (0, 1, 6, 7),  #face bottom
            ]


with pygmsh.geo.Geometry() as geom:
    for face in faces:
        geom.add_polygon([vertices[i] for i in face],
            mesh_size=5,
    )
    mesh = geom.generate_mesh()
```

Somehow resulted in my system path being reduced.

**Before**
```
PATH=PATH=D:\Programs\ImageMagick-7.0.11-Q16-HDRI;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Users\andre\AppData\Local\Programs\Python\Python38\Scripts\;C:\Users\andre\AppData\Local\Programs\Python\Python38\;C:\Users\andre\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\andre\AppData\Local\Programs\Python\Python39\;C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\andre\AppData\Roaming\npm\node_modules\windows-build-tools\node_modules\.bin;C:\Users\andre\AppData\Roaming\npm\node_modules\.bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Users\andre\AppData\Local\Programs\Python\Python37\Scripts\;C:\Users\andre\AppData\Local\Programs\Python\Python37\;C:\Users\andre\AppData\Local\Microsoft\WindowsApps;C:\Users\andre\AppData\Local\Pandoc\;C:\Users\andre\AppData\Local\Programs\Microsoft VS Code\bin;C:\Program Files\FFmpeg\bin;C:\Users\andre\AppData\Local\atom\bin;C:\Users\andre\AppData\Local\Microsoft\WindowsApps;C:\Users\andre\AppData\Roaming\npm;C:\Users\andre\AppData\Local\Programs\MiKTeX\miktex\bin\x64\;D:\Programs\pdf2svg\dist-64bits;
```

**After**
```
PATH=c:\users\andre\appdata\local\programs\python\python38\lib\site-packages\numpy\DLLs;D:\Programs\ImageMagick-7.0.11-Q16-HDRI;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Prog;c:\users\andre\appdata\local\programs\python\python38\
```

### PR DESCRIPTION
**Pull request by AndrewMicallef**
My attempt to fix up the mess of my previous pull request. Fixes #434.
Replaces the hot mess that is #435.

The names have been changed and default values added inline with @nschloe 's comment [here](https://github.com/nschloe/pygmsh/issues/435#issuecomment-792361504_)

### CODE REVIEW

File: pygmsh/common/geometry.py [hunk 1]
```diff
@@ -39,7 +40,13 @@ def __init__(self, env):
         self._OUTWARD_NORMALS = []
 
     def __enter__(self):
-        gmsh.initialize()
+
+        if self.init_argv is None:
+            init_argv = []
+        else:
+            init_argv = self.init_argv
```
Comment (nschloe) [inline pygmsh/common/geometry.py:47]: This can be a one-liner
```
gmsh.initialize([] if self.init_argv is None else self.init_argv)
```

Comment (AndrewMicallef) [reply pygmsh/common/geometry.py:47]: Are you able to make that change?
I won't have time for this until next weekend.

Comment (AndrewMicallef) [reply pygmsh/common/geometry.py]: Okay, made that edit, no trouble.

**Comment by nschloe** [comment]
Thanks!

**Pull request merged**